tools/src/wyvern/target/corewyvernIL/metadata/HasMetadata.java

```diff
@@ -0,0 +1,9 @@
+package wyvern.target.corewyvernIL.metadata;
+
+import java.util.Iterator;
+
+public interface HasMetadata {
+    public Metadata[] getMetadata();
```

Comment:
This is indeed a good thing to think about, but the Java collections library has a nice solution: Collections.unmodifiableSet()

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableSet(java.util.Set)

This addresses the semantic concern, and is probably also more efficient than extracting an array.  It's unfortunate that there's no ImmutableSet type in Java; it was considered but rejected as making the library too complex.